File: app/src/androidTest/java/io/github/fvasco/pinpoi/dao/PlacemarkCollectionDaoTest.java
package io.github.fvasco.pinpoi.dao;

import android.test.AndroidTestCase;
import android.test.RenamingDelegatingContext;

import org.junit.Test;

import java.util.List;

import io.github.fvasco.pinpoi.model.PlacemarkCollection;

/**
 * @author Francesco Vasco
 */
public class PlacemarkCollectionDaoTest extends AndroidTestCase {

    private PlacemarkCollectionDao dao;

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        dao = new PlacemarkCollectionDao(new RenamingDelegatingContext(getContext(), "test_"));
        dao.open();
        for (final PlacemarkCollection pc : dao.findAllPlacemarkCollection()) {
            dao.delete(pc);
        }
    }

    @Override
    protected void tearDown() throws Exception {
        dao.close();
        dao = null;
        super.tearDown();
    }

    @Test
    public void testFindPlacemarkCollectionCategory() throws Exception {
        PlacemarkCollection pc = new PlacemarkCollection();
        pc.setName("test");
        pc.setDescription("description");
        pc.setSource("source");
        pc.setCategory("CATEGORY");
        pc.setLastUpdate(System.currentTimeMillis());
        dao.insert(pc);

        List<String> list = dao.findAllPlacemarkCollectionCategory();
        assertEquals(1, list.size());
        String category = list.get(0);
        assertEquals("CATEGORY", category);
    }

    @Test
    public void testFindPlacemarkCollection() throws Exception {
        PlacemarkCollection pc = new PlacemarkCollection();
        pc.setName("test");
        pc.setDescription("description");
        pc.setSource("source");
        pc.setCategory("CATEGORY");
        pc.setLastUpdate(1);
        pc.setPoiCount(2);
        dao.insert(pc);

        final long id = pc.getId();
        // check findPlacemarkCollectionById
        pc = dao.findPlacemarkCollectionById(id);
        assertEquals(id, pc.getId());
        assertEquals("test", pc.getName());
        assertEquals("description", pc.getDescription());
        assertEquals("source", pc.getSource());
        assertEquals("CATEGORY", pc.getCategory());
        assertEquals(1, pc.getLastUpdate());
        assertEquals(2, pc.getPoiCount());

        pc.setName("test2");
        pc.setDescription("description2");
        pc.setSource("source2");
        pc.setCategory("CATEGORY2");
        pc.setLastUpdate(5);
        pc.setPoiCount(6);
        dao.update(pc);

        // check findAllPlacemarkCollection
        List<PlacemarkCollection> list = dao.findAllPlacemarkCollection();
        assertEquals(1, list.size());
        pc = list.get(0);
        assertEquals(id, pc.getId());
        assertEquals("test2", pc.getName());
        assertEquals("description2", pc.getDescription());
        assertEquals("source2", pc.getSource());
        assertEquals("CATEGORY2", pc.getCategory());
        assertEquals(5, pc.getLastUpdate());
        assertEquals(6, pc.getPoiCount());

        dao.delete(pc);
        list = dao.findAllPlacemarkCollection();
        assertTrue(list.isEmpty());
        assertNull(dao.findPlacemarkCollectionById(1));
    }

    public void testFindPlacemarkCollectionByName() throws Exception {
        PlacemarkCollection pc = new PlacemarkCollection();
        pc.setName("test");
        pc.setDescription("description");
        pc.setSource("source");
        pc.setCategory("CATEGORY");
        pc.setLastUpdate(5);
        pc.setPoiCount(6);
        dao.insert(pc);

        final long id = pc.getId();
        // check findPlacemarkCollectionByName
        pc = dao.findPlacemarkCollectionByName("test");
        assertEquals(id, pc.getId());
        assertEquals("test", pc.getName());
        assertEquals("description", pc.getDescription());
        assertEquals("source", pc.getSource());
        assertEquals("CATEGORY", pc.getCategory());
        assertEquals(5, pc.getLastUpdate());
        assertEquals(6, pc.getPoiCount());
    }
}

File: app/src/androidTest/java/io/github/fvasco/pinpoi/dao/PlacemarkCollectionDaoTest.kt
package io.github.fvasco.pinpoi.dao

import android.content.Context
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.platform.app.InstrumentationRegistry
import io.github.fvasco.pinpoi.model.PlacemarkCollection
import org.junit.After
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

/**
 * @author Francesco Vasco
 */
@RunWith(AndroidJUnit4::class)
class PlacemarkCollectionDaoTest {

    private lateinit var dao: PlacemarkCollectionDao

    lateinit var instrumentationContext: Context

    @Before
    fun setUp() {
        instrumentationContext = InstrumentationRegistry.getInstrumentation().context
        dao = PlacemarkCollectionDao(instrumentationContext)
        dao.open()
        for (pc in dao.findAllPlacemarkCollection()) {
            dao.delete(pc)
        }
    }

    @After
    fun tearDown() {
        dao.close()
    }

    @Test
    fun testFindPlacemarkCollectionCategory() {
        val pc = PlacemarkCollection()
        pc.name = "test"
        pc.description = "description"
        pc.source = "source"
        pc.category = "CATEGORY"
        pc.lastUpdate = System.currentTimeMillis()
        dao.insert(pc)

        val list = dao.findAllPlacemarkCollectionCategory()
        assertEquals(1, list.size)
        val category = list[0]
        assertEquals("CATEGORY", category)
    }

    @Test
    fun testFindPlacemarkCollection() {
        val pc = PlacemarkCollection()
        pc.name = "test"
        pc.description = "description"
        pc.source = "source"
        pc.category = "CATEGORY"
        pc.lastUpdate = 1
        pc.poiCount = 2
        dao.insert(pc)

        val id = pc.id
        // check findPlacemarkCollectionById
        val dbpc = dao.findPlacemarkCollectionById(id) ?: error("not found")
        assertEquals(id, dbpc.id)
        assertEquals("test", dbpc.name)
        assertEquals("description", dbpc.description)
        assertEquals("source", dbpc.source)
        assertEquals("CATEGORY", dbpc.category)
        assertEquals(1, dbpc.lastUpdate)
        assertEquals(2, dbpc.poiCount)

        pc.name = "test2"
        pc.description = "description2"
        pc.source = "source2"
        pc.category = "CATEGORY2"
        pc.lastUpdate = 5
        pc.poiCount = 6
        dao.update(pc)

        // check findAllPlacemarkCollection
        val list1 = dao.findAllPlacemarkCollection()
        assertEquals(1, list1.size)
        val pc0 = list1[0]
        assertNotNull(pc0)
        assertEquals(id, pc0.id)
        assertEquals("test2", pc0.name)
        assertEquals("description2", pc0.description)
        assertEquals("source2", pc0.source)
        assertEquals("CATEGORY2", pc0.category)
        assertEquals(5, pc0.lastUpdate)
        assertEquals(6, pc0.poiCount)

        dao.delete(pc)
        val list0 = dao.findAllPlacemarkCollection()
        assertTrue(list0.isEmpty())
        assertNull(dao.findPlacemarkCollectionById(1))
    }

    @Test
    fun testFindPlacemarkCollectionByName() {
        val pc = PlacemarkCollection()
        pc.name = "test"
        pc.description = "description"
        pc.source = "source"
        pc.category = "CATEGORY"
        pc.lastUpdate = 5
        pc.poiCount = 6
        dao.insert(pc)

        val id = pc.id
        // check findPlacemarkCollectionByName
        val dbpc = dao.findPlacemarkCollectionByName("test") ?: error("test not found")
        assertEquals(id, dbpc.id)
        assertEquals("test", dbpc.name)
        assertEquals("description", dbpc.description)
        assertEquals("source", dbpc.source)
        assertEquals("CATEGORY", dbpc.category)
        assertEquals(5, dbpc.lastUpdate)
        assertEquals(6, dbpc.poiCount)
    }
}